File: app/src/main/java/com/elegion/android/template/ui/splash/SplashView.java
package com.elegion.android.template.ui.splash;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

/**
 * @author mikhail.barannikov on 04.08.2017
 */
interface SplashView extends MvpView {
    @StateStrategyType(OneExecutionStateStrategy.class)
    void openLogin();

    @StateStrategyType(OneExecutionStateStrategy.class)
    void openFeatures();
}

File: app/src/main/java/com/elegion/android/template/ui/splash/SplashView.kt
package com.elegion.android.template.ui.splash

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

internal interface SplashView : MvpView {
    @StateStrategyType(OneExecutionStateStrategy::class)
    fun openLogin()

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun openFeatures()
}
